zk-sdk/src/encryption/elgamal.rs

```diff
@@ -33,22 +50,10 @@ use {
     },
     rand::rngs::OsRng,
     serde::{Deserialize, Serialize},
-    sha3::{Digest, Sha3_512},
-    solana_derivation_path::DerivationPath,
-    solana_sdk::{
-        signature::Signature,
-        signer::{
-            keypair::generate_seed_from_seed_phrase_and_passphrase, EncodableKey, EncodableKeypair,
-            SeedDerivable, Signer, SignerError,
-        },
-    },
-    std::{
-        convert::TryInto,
-        error, fmt,
-        io::{Read, Write},
-        path::Path,
-    },
+    sha3::Sha3_512,
+    std::{convert::TryInto, fmt},
     subtle::{Choice, ConstantTimeEq},
+    wasm_bindgen::prelude::*,
```

Comment:
Can we only pull this in for wasm builds, similar to https://github.com/anza-xyz/agave/blob/83e7d84bcc4cf438905d07279bc07e012a49afd9/sdk/pubkey/src/lib.rs#L31-L35